test/bad-arguments.test.ts

```diff
@@ -0,0 +1,317 @@
+import camelize from 'camelize';
+
+import config from '../lib/config';
+
+import csp = require('..');
+import { CSPOptions } from '../lib/types';
+
+const SOURCELIST_DIRECTIVES = [
+  'base-uri',
+  'child-src',
+  'connect-src',
+  'default-src',
+  'font-src',
+  'form-action',
+  'frame-ancestors',
+  'frame-src',
+  'img-src',
+  'manifest-src',
+  'media-src',
+  'object-src',
+  'prefetch-src',
+  'script-src',
+  'style-src',
+  'worker-src',
+];
+const SOURCELISTS_WITH_UNSAFES = ['script-src', 'style-src', 'worker-src'];
+const SOURCELISTS_WITH_STRICT_DYNAMIC = ['default-src', 'script-src'];
+const BOOLEAN_DIRECTIVES = ['block-all-mixed-content', 'upgrade-insecure-requests'];
+const PLUGINTYPE_DIRECTIVES = ['plugin-types'];
+const URI_DIRECTIVES = ['report-to', 'report-uri'];
+const REQUIRESRIFOR_DIRECTIVES = ['require-sri-for'];
+const SANDBOX_DIRECTIVES = ['sandbox'];
+
+function assertThrowsWithArg (arg: 'NO_ARGUMENTS' | CSPOptions, expectedMessage: string) {
```

Comment:
Could `NO_ARGUMENTS` instead be an enum or Symbol?

Reply:
Now a Symbol 👍